vector/v.in.dwg/main.c

```diff
@@ -137,8 +137,11 @@ int main(int argc, char *argv[])
     if (!adInitAd2(path, &initerror)) {
         sprintf(buf, _("Unable to initialize OpenDWG Toolkit, error: %d: %s."),
                 initerror, adErrorStr(initerror));
-        if (initerror == AD_UNABLE_TO_OPEN_INIT_FILE)
-            sprintf(buf, _("%s Cannot open %s"), buf, path);
+        size_t buflen = strlen(buf);
+        if (initerror == AD_UNABLE_TO_OPEN_INIT_FILE) {
+            snprintf(buf + buflen, 2000 - buflen - 1, (" Cannot open %s"),
```

Comment:
I recommend you take a closer look at the `snprintf()` manual or https://en.cppreference.com/w/c/io/fprintf to see how it works and specifically what arguments it can receive.
Furthermore, make a small test file and try it out to get to know it better.

Reply:
Thanks for the review @nilason!

I have made the following changes after looking at the `snprintf()` manual and testing few things out:

1. Using a variable of `size_t` datatype for `bufsz` argument in the function call.
2. Using `BUFSIZE - buflen` rather than `2000 - buflen - 1` as snprintf guarantees that it only writes `BUFSIZE-buflen-1` characters onto the buffer leaving space for null character.
3. There is a chance that prior to this `snprintf()` call, buffer can overflow from previous `sprintf()`, as we do not have boundary checks. I have modified to use `snprintf()` over there as well.

     Now, by the time we reach second `snprintf()` we are guaranteed that the buffer hasn't overflowed and we are writing at a valid space.
     
I checked with a small test [example](https://pastebin.com/GuCz39nF) that if we do this, we would be writing one string after another in the same buffer, which is what we are trying to achieve here.

Let me know what do you think and if there is something I missed. Thanks!

Reply:
Now the code looks good!